### ISSUE TITLE
End-to-end support for conditional syncing of data types in SyncManager

### ISSUE THREAD

**Issue by grigoryk** [mozilla-mobile/android-components#1584]
Currently we can disable syncing by simply "not syncing" an engine, but that's quite what needs to happen when we're working with Sync1.5. Turning off syncing of a data type is a global setting for the ecosystem and should affect other clients. We'll need appservices support for that.

**Comment by grigoryk** [comment]
This is largely blocked on support the app-services lib support. E.g. as part of this, we need to write to sync1.5 info/collections, which is the job of the upcoming Rust SyncManager - see mozilla/application-services#714.

**Comment by grigoryk** [comment]
Blocked on https://github.com/mozilla-mobile/android-components/issues/2727

**Comment by grigoryk** [comment]
At this point this is just a duplicate of #2727; closing.
